File: service/src/main/java/at/michaeladam/polar/service/kanban/view/WorkflowStatusView.java
package at.michaeladam.polar.service.kanban.view;

import at.michaeladam.polar.persistence.common.ID;
import at.michaeladam.polar.persistence.kanban.model.WorkflowStatus;
import lombok.*;
import lombok.experimental.Accessors;

import java.util.ArrayList;
import java.util.List;

@NoArgsConstructor
@Accessors(chain = true)
@Getter
@Setter
@ToString
@Builder
@AllArgsConstructor
public class WorkflowStatusView  {
    private ID<WorkflowStatusView> id;
    private String name;
    private String description;
    private List<IssueView> issues;
    private WorkflowStatus.WorkflowType workflowType;

    public List<IssueView> getIssues() {
        if(issues == null)
            issues = new ArrayList<>();
        return issues;
    }
}

File: resource/frontend/src/main/webapp/src/app/api/models/workflow-status-view.ts
/* tslint:disable */
/* eslint-disable */
import { IdWorkflowStatusView } from './id-workflow-status-view';
import { IssueView } from './issue-view';
import { WorkflowType } from './workflow-type';
export interface WorkflowStatusView {
  description?: string;
  id?: IdWorkflowStatusView;
  issues?: Array<IssueView>;
  name?: string;
  workflowType?: WorkflowType;
}
